packages/material-ui/src/Slider/Slider.test.js

```diff
@@ -168,32 +166,66 @@ describe('<Slider />', () => {
   describe('range', () => {
     it('should support keyboard', () => {
       const { getAllByRole } = render(<Slider defaultValue={[20, 30]} />);
-      const [thumb1, thumb2] = getAllByRole('slider');
+      const [slider1, slider2] = getAllByRole('slider');
+
+      act(() => {
+        slider1.focus();
+        fireEvent.change(slider1, { target: { value: '21' } });
+      });
+
+      expect(slider1.getAttribute('aria-valuenow')).to.equal('21');
+      expect(slider2.getAttribute('aria-valuenow')).to.equal('30');
+
+      act(() => {
+        slider2.focus();
+        fireEvent.change(slider2, { target: { value: '31' } });
+      });
+
+      expect(slider1.getAttribute('aria-valuenow')).to.equal('21');
+      expect(slider2.getAttribute('aria-valuenow')).to.equal('31');
 
       act(() => {
-        thumb1.focus();
-        fireEvent.keyDown(thumb1, {
-          key: 'ArrowRight',
-        });
+        slider1.focus();
+        fireEvent.change(slider1, { target: { value: '31' } });
       });
 
-      expect(thumb1.getAttribute('aria-valuenow')).to.equal('21');
+      expect(slider1.getAttribute('aria-valuenow')).to.equal('31');
+      expect(slider2.getAttribute('aria-valuenow')).to.equal('31');
+      expect(document.activeElement).to.have.attribute('data-index', '0');
 
       act(() => {
-        thumb2.focus();
-        fireEvent.keyDown(thumb2, {
-          key: 'ArrowLeft',
-        });
+        slider1.focus();
+        fireEvent.change(slider1, { target: { value: '32' } });
       });
 
-      expect(thumb2.getAttribute('aria-valuenow')).to.equal('29');
+      expect(slider1.getAttribute('aria-valuenow')).to.equal('31');
+      expect(slider2.getAttribute('aria-valuenow')).to.equal('32');
+      expect(document.activeElement).to.have.attribute('data-index', '1');
     });
 
     it('should focus the slider when dragging', () => {
-      const { getByRole } = render(<Slider defaultValue={30} step={10} marks />);
-      const thumb = getByRole('slider');
-      fireEvent.mouseDown(thumb);
-      expect(thumb).toHaveFocus();
+      const { getByRole, getByTestId, container } = render(
+        <Slider
+          componentsProps={{ thumb: { 'data-testid': 'thumb' } }}
+          defaultValue={30}
+          step={10}
+          marks
+        />,
+      );
+      const slider = getByRole('slider');
+      const thumb = getByTestId('thumb');
+
+      stub(container.firstChild, 'getBoundingClientRect').callsFake(() => ({
+        width: 100,
+        left: 0,
+      }));
```

Comment:
We also run tests in real browsers where we shouldn't need this. You can look at existing tests how we determine whether we run in JSDOM or not.

Reply:
This was the approach this file was using originally in order to setup and simulate mouse movements since the code uses `const { width, height, bottom, left } = slider.getBoundingClientRect();` to determine what the new value should be in relation to where the users finger is. I borrowed this from the existing code, as a simple way to ensure the width was 100 in order to keep the translation to a value clear (math is a little simpler and easier to spot this way). That being said I can look at updating those old tests or seeing if just working with the width the slider is getting naturally for the new tests.

Reply:
Makes sense. I'll follow-up after this is merged to make sure we're simulating actual layout. We shouldn't need to simulate it in our browser test suite since those run headful i.e. with layout.